dashboard/src/components/Catalog/Catalog.tsx

```diff
@@ -220,6 +237,58 @@ function Catalog(props: ICatalogProps) {
         intersection(filters[filterNames.CATEGORY], getOperatorCategories(c)).length,
     );
 
+  // avoid repeating the same requests with the same params with useCallback
+  // avoid overloading the backend with a max rate of 2 reqs/second (= debounce 0.5s)
+  const debouncedFetchCharts = useCallback(
+    debounce(() => {
+      fetchCharts(cluster, namespace, reposFilter, page, size, records, searchFilter);
+    }, 500),
+    [fetchCharts, cluster, namespace, reposFilter, page, size, records, searchFilter],
+  );
+
+  const observeBorder = useCallback(
+    // Check if the IntersectionAPI is enabled
+    node => {
+      if (
+        "IntersectionObserver" in window &&
+        "IntersectionObserverEntry" in window &&
+        "isIntersecting" in window.IntersectionObserverEntry.prototype
+      ) {
+        if (node !== null) {
+          // https://developer.mozilla.org/en-US/docs/Web/API/IntersectionObserver
+          new IntersectionObserver(
+            entries => {
+              // https://developer.mozilla.org/en-US/docs/Web/API/IntersectionObserverEntry
+              entries.forEach(entry => {
+                if (
+                  entry.isIntersecting &&
+                  // Disable scrolling when only operators are selected
+                  (!filters[filterNames.TYPE].length ||
+                    filters[filterNames.TYPE].find((type: string) => type === "Charts")) &&
+                  // Disable scrolling if all the charts have been fetched
+                  // && !isFetching
+                  !hasFinished &&
+                  // Enable scrolling just if every parameter is already updated
+                  clusterUpdate.current === cluster &&
+                  namespaceUpdate.current === namespace &&
+                  reposFilterUpdate.current === reposFilter &&
+                  searchFilterUpdate.current === searchFilter
```

Comment:
I don't understand these conditions, I believe is trying to avoid duplicated calls but can you elaborate?

Reply:
Yes, I'd want to explain it a little bit more as it is just another attempt to avoid dups:

Whenever a change occurs and it is detected by the `useEffect`, I (try to) prevent these values to be actually updated before triggering the reset. It is being done by means of the `useRef`. The good point here is that mutating the `.current` won't cause a re-render.

If we remove these conditions, duplicated requests would appear.

Reply:
that's fine then but as above, you'd need to ensure this with a test. First, comment this code, and write a test ensure it's failing (there are more than one request for the same params), then enable the functional code and the test should pass.